### ISSUE TITLE
Other (not-implemented) metrics

### ISSUE THREAD

**Issue by marcosci** [r-spatialecology/landscapemetrics#15]
To keep it a bit more organized, here a single list with the metrics we are currently missing.

## Cell-aggregation metrics

- [ ] https://github.com/r-spatialecology/landscapemetrics/issues/288

## Contrast metrics

- [ ] https://fragstats.org/index.php/fragstats-metrics/contrast-metrics

## Patch level
- [ ] Proximity index (Responsible: :ghost:) -> Needs user to specify search radius
- [ ] Similarity index (Responsible: :ghost:) -> Needs user to input similarity table

## Class level
- [ ] Linearity index distribution (Responsible: :ghost:) -> Not in FRAGSTATS manual
- [ ] Connectance (Responsible: @marcosci ) -> Needs nearest neighbor distances, so we wait until we have a good solution for ENN
- [ ] Proximity index distribution (Responsible: :ghost:) -> Needs search radius
- [ ] Similarity index distribution (Responsible: :ghost:) -> Needs similarity table
- [ ] Linearity index distribution (Responsible: :ghost:) -> Not in FRAGSTATS manual
- [ ] MFRAC (Responsible: :ghost:) 

## Landscape level
- [ ] Linearity index distribution (Responsible: :ghost:) -> Missing in FRAGSTATS manual
- [ ] Proximity index distribution (Responsible: :ghost:) -> Needs search radius
- [ ] Similarity index  (Responsible: :ghost:) -> Needs similarity table
- [ ] Connectance (Responsible: @marcosci )

**Comment by marcosci** [comment]
MFRAC is missing, too:

http://www.umass.edu/landeco/research/fragstats/documents/Metrics/Contagion%20-%20Interspersion%20Metrics/Metrics/C118%20-%20MFRAC.htm

**Comment by gazzapol** [comment]
all contrast metrics seems missing

**Comment by mhesselbarth** [comment]
Yes, at the moment we do not include them and probably won't be in the nearer future.

**Comment by Klemet** [comment]
Hello everybody !

Just wanted to say that it seems that the metric CAI_AM is not implemented in your package yet; it seems important however, seems it's one of the landscape metrics proposed by Wang *et al.* (2014) because it has the following properties : 
- It's not very correlated with habitat abundance
- It has a good correlation with habitat fragmentation/aggregation
- It's relatively easy to interpret

I quote from their article : "*Among the metrics measuring CORE AREA, Area Weighted Mean Core Area Index (CAI_AM) is perhaps the most intuitive to interpret as it is the only metric that is not a measure of variance. For this reason, we recommend CAI_AM as a broadly appropriate landscape fragmentation measurement index for measuring core area.*"

It should be relatively easy to implement, as if I understand correctly, it corresponds to the total core area for the class (or the landscape) divided by the total area for the class (or the landscape) and multiplied by 100 to get a percentage. I hope I understand it right in saying this, as there is no formula in the FRAGSTAT documentation for this one.

I quote from the FRAGSTAT documentation : "*Similarly, at the class and landscape levels core area index area-weighted mean (CAI_AM) quantifiescore area for the entire class or landscape as a percentage of total class or landscape area,respectively. Note, that this is equivalent to the total core area index reported in FRAGSTATS 2.0.*".

I hope I'm not mistaken in what I say; if I can try to implement it with the existing functions of the package, I'd be delighted to. I'm just not very familiar with the process of proposing modifications as an outsider of the repository.

Reference : Wang, X., Blanchet, F. G. et Koper, N. (2014). Measuring habitat fragmentation: An evaluation of landscape pattern metrics. Methods in Ecology and Evolution, 5(7), 634‑646. doi: 10.1111/2041-210X.12198

**Comment by mhesselbarth** [comment]
Hey, 

You are correct, we don't have any area-weighted means at the moment for any metric. However, if you go to our homepage, you can find a short example how to calculate them yourself. Here is the link: [Landscape distribution statistics](https://r-spatialecology.github.io/landscapemetrics/articles/articles/landscape_distribution_statistics.html)

**Comment by Klemet** [comment]
Thanks a bunch, Maximilian ! I was able to compute CAI_AM really easily thanks to your help : ). I'd like to add that your documentation website is just awesome.

**Comment by aavotins** [comment]
One of the advantages of Fragstats is the ability to define different interactions between classes when calculating, e.g. core areas or edge contrasts. It seems that {landscapemetrics} is currently only able to use a single value instead of a (contrast oor edge depth) matrix. For example, functions lsm_c_tca, lsm_p_core. Is it planned (and is there any idea of timing?) to implement edge depth and contrast matrix metrics between classes? Or pmaybe there are some building blocks for this, that I have missed?

**Comment by mhesselbarth** [comment]
All edge metrics have a `edge_depth` which allows to specify how many rows of cells are considered "edge". You can use `show_cores(x, edge_depth=1)` vs.  `show_cores(x, edge_depth=3)` to get a visualization of this. 

As for the contrast matrix, I don't think this will be implemented soon due to missing time.

**Comment by toddellis** [comment]
Hello, I'm wondering if there's a possible mistake regarding landscapemetrics::list_lsm() labeling the contagion index as the connectance index in the name column?

See:

`landscapemetrics::list_lsm() |> dplyr::filter(name == "connectance")`

(Apologies if I'm in the wrong place -- I have never opened an issue or commented on a GitHub repo -- but I see the connectance index here as still not implemented.)

**Comment by mhesselbarth** [comment]
You are correct. I just opened a new issue #345  to keep track of this. Will be fixed
